go/oasis-node/cmd/common/metrics/mem.go

```diff
@@ -0,0 +1,98 @@
+// Package metrics implements a prometheus metrics service.
+package metrics
+
+import (
+	"fmt"
+	"os"
+	"sync"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/procfs"
+)
+
+const (
+	MetricMemVmSizeBytes   = "oasis_worker_mem_VmSize_bytes" // nolint: golint
+	MetricMemRssAnonBytes  = "oasis_worker_mem_RssAnon_bytes"
+	MetricMemRssFileBytes  = "oasis_worker_mem_RssFile_bytes"
+	MetricMemRssShmemBytes = "oasis_worker_mem_RssShmem_bytes"
+)
+
+var (
+	vmSizeGauge   prometheus.Gauge // nolint: golint
+	rssAnonGauge  prometheus.Gauge
+	rssFileGauge  prometheus.Gauge
+	rssShmemGauge prometheus.Gauge
```

Comment:
All of these could (should?) be moved into service structs? (similar for other services)

Reply:
Why? These are global (per process) anyway so it doesn't make sense having multiple instances.

Reply:
There will be only one instance of the worker anyway, and that way these references are not be available in the whole `metrics` package.

Reply:
But we use global metrics everywhere else AFAIK?

Reply:
We do, on the other hand the prom. client lib doesn't: https://github.com/prometheus/client_golang/blob/master/prometheus/process_collector.go#L25-L29

Either way is fine so it's just a preference, feel free to keep it as it is.

Reply:
For consistency with other metrics in the code, I left it global.